### PR TITLE
ENH: Separate "external" and "internal" configuration

### PR DESCRIPTION
**Pull request by davewalter**
## WHAT is this change about?
The goal of this change is to separate the "external" user-facing configuration from the "internal" intra-platform configuration required for CF for K8s to operate. It isolates the internal configuration in the `generate-internal-values.sh` script that will eventually be obsoleted by the implementation of a cluster-based secret management system (probably Quarks) and provides clear documentation and samples for the external configuration that the user is expected to provide when deploying cf-for-k8s.

One of my goals was to provide a working sample external values file for deploying locally on KinD, but that would have meant including real system and workloads certificates for vcap.me, so I decided against that.

## Does this PR introduce a change to `config/values.yml`?
No

## Acceptance Steps
Deploy and run smoke-tests.

### CODE REVIEW

**Comment by davewalter** [comment]
@jamespollard8 @paulcwarren @ericpromislow Not sure the "real" vcap.me cert in the sample file for kind is a good idea. I got a warning email from Github about it.

File: docs/deploy.md [hunk 1]
```diff
@@ -68,34 +66,21 @@ Currently, we test the following two container registries:
...
    git clone https://github.com/cloudfoundry/cf-for-k8s.git
    cd cf-for-k8s
-   TMP_DIR=<your-tmp-dir-path> ; mkdir -p ${TMP_DIR}
+   VALUES_DIR=<your-values-dir-path> ; mkdir -p ${VALUES_DIR}
    ```
 
 1. Create a "CF Installation Values" file and configure it<a name="cf-values"></a>:
 
-   You can either: a) auto-generate the installation values or b) create the values by yourself.
+     1. Clone one of the sample values files from the `sample-cf-install-values` directory to use as a starting point.
```
Comment (jamespollard8) [inline docs/deploy.md:74]: Maybe "Clone an appropriate sample values file from the..."

File: docs/deploy.md [hunk 2]
```diff
@@ -143,18 +128,26 @@ Currently, we test the following two container registries:
...
+   ./hack/generate-internal-values.sh --values-file ${VALUES_DIR}/cf-values.yml > ${VALUES_DIR}/cf-internal-values.yml
+   ```
+
 1. Run the following commands to install Cloud Foundry on your Kubernetes cluster:
 
       i. Render the final K8s template to raw K8s configuration
 
          ```console
-         ytt -f config -f ${TMP_DIR}/cf-values.yml > ${TMP_DIR}/cf-for-k8s-rendered.yml
+         ytt -f config -f ${VALUES_DIR}/cf-values.yml -f ${VALUES_DIR}/cf-internal-values.yml > ${VALUES_DIR}/cf-for-k8s-rendered.yml
```
Comment (jamespollard8) [inline docs/deploy.md:144]: the fact that we save the rendered k8s yaml here means that maybe we should keep the variable as TMP_DIR? (or use both a values dir and a tmp dir)

File: docs/platform_operators/external-databases.md [hunk 3]
```diff
@@ -166,15 +166,15 @@ In the following section, we will show how to setup an AWS RDS database and conf
...
 1. [Install cf-for-k8s](../deploy.md)
 
-    i. Configure your [`cf-values.yml`](../deploy.md#cf-values) file
-    i. Render the final K8s template to raw K8s yaml. Pass the `"$DB_VALUES_FILE"` file as additional file to `ytt`
+    i. Configure your [`cf-values.yml`](../deploy.md#cf-values) file.
+    i. Render the final K8s template to raw K8s yaml. Pass the `"${DB_VALUES_FILE}"` file as additional file to `ytt`:
 
     ```bash
-    ytt -f config -f "$TMPDIR/cf-values.yml" -f "$DB_VALUES_FILE" > "$TMPDIR/cf-for-k8s-rendered.yml"
+    ytt -f config -f "${VALUES_DIR}/cf-values.yml" -f "${DB_VALUES_FILE}" > "${VALUES_DIR}/cf-for-k8s-rendered.yml"
```
Comment (jamespollard8) [inline docs/platform_operators/external-databases.md:173]: should we leave the TMP_DIR in this file as it was?

**Comment by davewalter** [comment]
I'm not sure it is worth trying to keep this PR alive. The implementation of Quarks secret management will solve many of the same problems that I tried to solve, and it will be much easier to revisit this work once that is incorportated. I am going to close this PR out.

**Comment by jamespollard8** [comment]
> I'm not sure it is worth trying to keep this PR alive. The implementation of Quarks secret management will solve many of the same problems that I tried to solve, and it will be much easier to revisit this work once that is incorporated. I am going to close this PR out.

Makes sense - thanks Dave

**Pull request closed without merging**